tests/test_codegen_directive.cpp

```
@@ -470,6 +552,86 @@ TEST_F(DirectiveTest, NativeFnLibraryDoesNotAffectDispatch) {
+TEST_F(DirectiveTest, NativeFnLibraryGenericDispatchBool) {
+    // Tests BoolFromI64 wrapping in the generic dispatch path.
+    std::string output = runSource(
+        "@native(\"path\")\n"
+        "function is_absolute(p: str) -> bool\n"
+        "print(is_absolute(\"/usr/bin\"))\n"
+        "print(is_absolute(\"relative\"))\n"
+    );
+    EXPECT_EQ(output, "true\nfalse\n");
+}
+
+TEST_F(DirectiveTest, NativeFnLibraryGenericDispatchResultBool) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)